pkg/apis/storage/types.go

```diff
@@ -293,8 +293,45 @@ type CSIDriverSpec struct {
 	// "csi.storage.k8s.io/pod.uid": string(pod.UID)
 	// +optional
 	PodInfoOnMount *bool
+
+	// VolumeModes defines what kind of volumes this CSI volume driver supports.
+	// The default if the list is empty is "Persistent", which is the usage
+	// defined by the CSI specification and implemented in Kubernetes via the usual
+	// PV/PVC mechanism.
+	// The other mode is "Ephemeral". In this mode, volumes are defined inline
+	// inside the pod spec with CSIVolumeSource and their lifecycle is tied to
+	// the lifecycle of that pod. A driver has to be aware of this
+	// because it is only going to get a NodePublishVolume call for such a volume.
```

Comment:
I can add it, but I have no idea how self-contained the API documentation is supposed to be. We already started adding a lot of text here that duplicates information found elsewhere. If we now continued with adding links, we might as well remove the duplicate text and just say "for more information, see ...".

Reply:
We are trying to strive towards having the API documentation be very descriptive so that a user reading it without much background on the subject can get a decent idea what the API is supposed to do.

So I think the comments here are good, and then for further information on how to actually implement this feature in your csi driver, we can refer to the csi driver dev docs.

Reply:
Done.